python/jiminy_pywrap/src/engine.cc

```diff
@@ -277,16 +277,24 @@ namespace jiminy::python
             {
                 aInit.emplace(convertFromPython<std::map<std::string, Eigen::VectorXd>>(aInitPy));
             }
-            std::optional<AbortSimulationFunction> callback;
+            AbortSimulationFunction callback;
```

Comment:
I did this change because the callback was not sent to `simulate`. The progress bar did not progress. If I revert this change and I still want to send the callback, I have :
```
error: no matching function for call to ‘jiminy::Engine::simulate(double&, std::enable_if_t<true, std::map<std::__cxx11::basic_string<char>, Eigen::Matrix<double, -1, 1>, std::less<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, Eigen::Matrix<double, -1, 1> > > > >, std::enable_if_t<true, std::map<std::__cxx11::basic_string<char>, Eigen::Matrix<double, -1, 1>, std::less<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, Eigen::Matrix<double, -1, 1> > > > >, std::optional<std::map<std::__cxx11::basic_string<char>, Eigen::Matrix<double, -1, 1>, std::less<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, Eigen::Matrix<double, -1, 1> > > > >&, std::optional<std::function<bool()> >&)’
  285 |             return self.simulate(
      |                    ~~~~~~~~~~~~~^
  286 |                 endTime,
      |                 ~~~~~~~~         
  287 |                 convertFromPython<std::map<std::string, Eigen::VectorXd>>(qInitPy),
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  288 |                 convertFromPython<std::map<std::string, Eigen::VectorXd>>(vInitPy),
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  289 |                 aInit,
      |                 ~~~~~~           
  290 |                 callback);
      |                 ~~~~~~~~~        
compilation terminated due to -Wfatal-errors.
```